src/banana/navigator/navigator.py

```diff
@@ -232,12 +233,14 @@ def show_full_logs(self, t_fields=None, o_fields=None, keep_hashes=False):
         if len(theories_df) > 0:
             theories = theories_df[["hash"] + t_fields]
             theories = theories.rename(columns={"hash": "theory"})
+            theories["theory"] = theories["theory"].apply(lambda h: h[: self.hash_len])
         else:
             theories = theories_df
         ocards_df = pd.DataFrame(self.get_all(o))
         if len(theories_df) > 0:
             ocards = ocards_df[["hash"] + o_fields]
             ocards = ocards.rename(columns={"hash": "ocard"})
+            ocards["ocard"] = ocards["ocard"].apply(lambda h: h[: self.hash_len])
```

Comment:
I propose to solve the other way round: add a parameter to `list_all()` not to cut the hash at all (by default of course should cut).
This might become handy if ever there will be a conflict with the cut hashes.

Reply:
this turns out to be more complicated, since the hash cutting is done in the client programs: e.g. https://github.com/N3PDF/yadism/blob/ecc16f965504435423e44868790cf5d154e0d7b2/src/yadmark/navigator/navigator.py#L121 do we really want to propagate this information all around or just rely on `app.hash_len` to be adjusted once by the user?

Reply:
In the first place, it is done on two sides, so I would keep going with the internal one at least
https://github.com/N3PDF/banana/blob/179f4bc7499ae79a4681ad0bcaa0d21fdc2de9e3/src/banana/navigator/navigator.py#L189

For the other piece I'd say we should propagate as well, but we can postpone.

The principle is that hashes should be cut only in the UI, but never internally. The problem arose because `get_all` functions where UI in the first place, but they have been reused internally to build the `logs()` function.
The main purpose of those function is to define a list of relevant fields, so in the long run I'd split the list definition (as data attribute of navigator subclass) from the UI function, and recompute the tables directly in the logs (instead of abusing tables already produced for the end user).